### ISSUE TITLE
PetShop: Use HttpClient instead of HttpURLConnection

### ISSUE THREAD

**Issue by 21aslade** [softwareconstruction240/softwareconstruction#169]
Currently, PetShop's `ServerFacade` uses `HttpURLConnection` to make requests. `HttpURLConnection`'s API is rather unintuitive and bulky, and JDK 11 provided `HttpClient` as a replacement.

Switching to `HttpClient` makes the requests simpler and more representative of a modern approach to http. It would also close #157, #160, and #163, as`HttpClient`'s api does not differentiate between the input stream and the error stream.

I'm happy to provide the code/docs changes! They're all relatively simple. I just wanted to make an issue before making a PR to give some heads up.

**Comment by frozenfrank** [comment]
@21aslade Thank you for submitting this issue. Thank you for concisely summing up the problem and proposing a meaningful solution. You've done an amazing job providing links to our prior efforts to address the issue.

Additional research confirms that this proposal is legit. See the related resources:
* [Baeldung Introduction to changes](https://www.baeldung.com/java-9-http-client)
* [Reddit Discussion about Java HTTP clients](https://www.reddit.com/r/java/comments/1986h01/comment/ki5dxi2/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)
* [OpenJDK Http Examples](https://openjdk.org/groups/net/httpclient/recipes.html)
* [Oracle HttpClient docs](https://docs.oracle.com/en/java/javase/21/docs/api/java.net.http/java/net/http/HttpClient.html)

Personally, I like the upgrade approach a lot more than our previous attempts. I would love to see a PR that performs the upgrade. Recognizing that I have no actual authority, you may choose to wait for a more meaningful approval from @leesjensen or @jerodw.

**Comment by leesjensen** [comment]
Love it! We just need to update the following:

- Web API instruction
- Phase 3 instruction
- Petshop
- Slides
- Videos

Totally worth doing though. I thought I remembered an easier way to do client requests in Java!

@rodham @jerodw - what do you think?

**Comment by jerodw** [comment]
I agree. This looks like a great change. It isn't trivial though with the need to update slides, videos and assignment instructions. I think we should do this along with the change from Spark to Javelin over the summer.
